### ISSUE TITLE
Training with ResNet-18

### ISSUE THREAD

**Issue by hgaiser** [rbgirshick/py-faster-rcnn#200]
I created a Faster-RCNN ResNet-18 architecture based on the paper and the ResNet-50 deploy prototxt.

Visualization: https://ethereon.github.io/netscope/#/gist/543ea0a62f0596d56d9ba8ccfce62599
Train prototxt: https://gist.github.com/hgaiser/543ea0a62f0596d56d9ba8ccfce62599

I was training this network on roughly 25.000 images for 180.000 iterations, starting with a learning rate of 0.1 and decreasing every 50.000 iterations by 10. I have no pretrained model for ResNet-18 so I start the network from scratch.

Solver prototxt: https://gist.github.com/hgaiser/362d9cee8a0bf596924700dbf5253cdd

Training does not appear to converge however, not even a bit. Results on my validation set gives a mAP of 0.0002. Using VGG16 with a network pretrained on ImageNet training does nicely converge, so the data seems to be fine. Any thoughts on why training on ResNet-18 does not converge?

**Comment by happyharrycn** [comment]
My answer is that you simply need more data to train a 18-layer network from scratch. In the meanwhile, you can always use a pre-trained ResNet18 model from [here](https://github.com/facebook/fb.resnet.torch/tree/master/pretrained)

**Comment by hgaiser** [comment]
Thanks for your reply. I know about the tool which can convert torch models to caffe models, but how do I ensure that the same name for layers are used (and that I indeed made the correct architecture) ?

**Comment by happyharrycn** [comment]
@hgaiser You can write a simple test. Given the same input image, the torch version and the caffe version should give you the same output (as well as intermediate response) within the numerical tolerance.

**Comment by hgaiser** [comment]
Thank you again, I will try so when I have some more time on my hands (never worked with Torch yet so it's not that simple for me ;) )

**Comment by mengzhangjian** [comment]
@hgaiser ,How do you solve the problem of caffe version?

**Comment by xuewenyuan** [comment]
@hgaiser I met the same problem. Have you solve it?

**Comment by hgaiser** [comment]
Nope I gave up on ResNet-18 and used ResNet-50 with the pretrained networks with a generated prototxt from https://github.com/XiaozhiChen/resnet-generator .

**Comment by xuewenyuan** [comment]
alright, is it because the architecture of ResNet-18? The memory of my GPU is only 6G, which cannot run ResNet-50 😂

**Comment by hgaiser** [comment]
Hehe, alright. Not sure why it wasn't working for me, the deadline passed so we settled on VGG-16 quality, which was more than sufficient at the time.

**Comment by xuewenyuan** [comment]
anyway, thanks a lot
